examples/modules/example_module.cpp

```diff
@@ -65,27 +65,24 @@ class MyModule : public GenericService::Service, public ComponentBase {
 };
 
 int MyModule::which = 0;
-std::shared_ptr<ModuleService_> my_mod;
-
-void signal_handler(int signum) {
-    my_mod->close();
-}
 
 int main(int argc, char** argv) {
     if (argc != 2) {
         throw "need socket path as command line argument";
     }
 
-    // TODO(RSDK-1920) This is still causing non-graceful shutdown. Figure out why, and fix.
-    struct sigaction sig_handler;
-    sig_handler.sa_handler = signal_handler;
-    sigaction(SIGTERM, &sig_handler, nullptr);
-    sigemptyset(&sig_handler.sa_mask);
-    sig_handler.sa_flags = 0;
+    // C++ modules must handle SIGINT and SIGTERM. Make sure to create a sigset
```

Comment:
The comment is good, but it actually doesn't speak to one of the most important things here, which is the call to `pthread_sigmask` happening as early as possible. It is important to do this more or less at the beginning of main before other threads are created so that the new threads also inherit the mask. I'd suggest updating the comment to include that information to make it clear that the placement is not arbitrary.

Reply:
Great point; elaborated on necessary positioning of this code.